netbox_diode_plugin/api/authentication.py

```diff
@@ -65,6 +66,16 @@ def _introspect_token(self, token: str):
             return None
 
         if data.get("active"):
+            # if the plugin is configured to require specific token audience(s),
+            # reject the token if any are missing.
+            required_audience = get_required_token_audience()
+            if len(required_audience) > 0:
+                token_audience = set(data.get("aud", []))
+                for aud in required_audience:
+                    if aud not in token_audience:
+                        logger.error(f"Token audience {aud} not found in {token_audience}")
+                        return None
```

Comment:
nit: we can utilize set logic for subtraction to make it more pythonic like:
```suggestion
            if required_audience:
                token_audience = set(data.get("aud", []))
                missing_audience = set(required_audience) - token_audience
                if missing_audience:
                    logger.error(f"Token audience(s) {missing_audience} not found in {token_audience}")
                    return None
```
```